### ISSUE TITLE
make `plplot` an optional dependency

### ISSUE THREAD

**Issue by soupault** [owlbarn/owl#31]
Hello!
Do you think if there is a possibility to make `plplot` dependency optional?
Currently, installation of `plplot` OCaml module (and, thus, `owl`) terminates with an error if the user does not have `plplot` software installed.
`plplot` pre-build packages aren't provided for many of the popular distros (for example, in Archlinux the package is provided in AUR only), so it might be troublesome to ask the users to perform an extra step.

**Comment by ryanrhymes** [comment]
Hi @soupault , my apologies for the late reply. This is definitely doable since there are no core components in Owl really depend on Plot module. Having Plot module included is because we aim to provide a more integrated and complete environment for fast prototyping for numerical developers.

I think one possible solution can be conditional build, which I will look into later. Currently, you have to edit _oasis manually to remove owl_plot by yourself if you don't want this module.

**Comment by soupault** [comment]
@ryanrhymes Thanks, I see. From my point of view, applying conditional build will be the best choice.
In the meantime, I'd suggest to enrich README.md with the link to `plplot` installation guide (as it is not as simple as `opam install plplot`). This one should do - http://plplot.sourceforge.net/downloads.php . Would you agree?

**Comment by ryanrhymes** [comment]
I agree. I will add some explanation in readme about this later.

**Comment by nielmond** [comment]
Tried installing owl and got blocked because of this. I'm further confused because I installed plplot using `opam depext plplot.5.11.0` on OSX 10.12.6, and running that a second time returns with `All required OS packages found.` However, when trying to install just plplot afterwards I get `Failure: No valid installation of plplot or plplotd found.` I do see that it install plplot version 5.13.0_5. Is that the problem?

Running depext for `conf-openblas` (which also initially caused the installation to fail) seemed to work.
